### PR TITLE
Fix calculating of the row height in 7.2.x

### LINKED ISSUE #5012: There is an empty space in the treeGrid when hide a column with summaries
## Description  
There is an empty space in the treeGrid when hide a column with summaries

 * igniteui-angular version: 7.2
 * browser: n/a 

## Steps to reproduce  

1. Open dev sample treeGridFlatData
2. Click button rootAndChildLevels
3. Hide column Salary
4. Scroll to button

Note: The issue appears only in 7.2.x, in the 7.3.x this is fixed with the PR #4730

## Result  
There is an empty space in the treeGrid

## Expected result  
There should be no empty space

## Attachments  
Attach a sample if available, and screenshots, if applicable.

### PR DESCRIPTION
**Pull request by nrobakova**
Closes #5012  

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts [hunk 1]
```diff
@@ -1297,6 +1297,8 @@ export class IgxGridForOfDirective<T> extends IgxForOfDirective<T> implements On
             size = parseInt(this.igxForItemSize, 10) || 0;
             if (item && item.summaries) {
                 size = item.max;
+            } else if (item && item.groups && item.height) {
+                size = item.height;
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts]: this requires that the height is assigned to the ```IGroupByRecord``` by the ```groupDataRecursive<T>``` method, which is not the case in 7.2.x. You should get that code as well from the 7.3.x PR

Comment (nrobakova) [reply projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts]: @ChronosSF I updated the groupDataRecursive<T> to get the height as in the other PR

Comment (skrustev) [reply projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts]: In general for this related issue only the fix on line 670 would be enough.
The rest is for the group rows sizing when the regular rows have different height and so far it only happens for Multi Row Layout and it is not related to the IgxTreeGrid. Unless you want to bring the whole fix to 7.2.x as well.

**Comment by ddincheva** [comment]
the issue is resolved, but it is necessary tests to be updated

**Pull request merged**